docs/guidance/publishing-from-portal.md

```diff
@@ -1,9 +1,98 @@
+
+      The [field level mapping template](https://docs.google.com/spreadsheets/d/1xHLf_w193pp97zfzhLc_LI-yEXrR_eyscga06Qo1blk/copy) can be used to document your mapping.
+
+```
+
+<!-- TODO: Update link to mapping template -->
+
+Your mapping might identify:
+
+* **Gaps in your data** where data in the project level data specification is not currently collected or disclosed in your system. Use the specification as a guide to the information that is important to users and consider whether your system and business processes could be updated to collect and publish additional information.
+
+* **Gaps in the data specification** where data is collected by your system but doesn’t map to the specification. Rather than being excluded from your publication, such information should be included as additional fields in your data. Refer to [extending the schema](../projects/schema.md#extending-the-schema) for information on including additional fields in your data.
+
+
+### (2b) Map detailed contracting process data
+
+If you collect detailed data on contracting processes, refer to the [OCDS implementation journey](https://www.open-contracting.org/implement/#/2) for information and resources about mapping and publishing your contracting data using OCDS.
+
+### (3) Build your data, systems and processes
+
+Create a project level JSON file for each project your system has information on and use a [JSON schema validator](https://json-schema.org/implementations.html#validators) to check that the files are structurally correct against the project level data specification.
+
+If you are also publishing contracting data using OCDS, create an OCDS release each time the data about a contracting process changes and use the [OCDS data review tool](http://standard.open-contracting.org/review/) to check your OCDS releases.
+
+Make sure you have systems and/or business processes in place to keep the data you produce up to date.
+
+### (4) Publish your data
+
+Publish your project level JSON fields (as either static files or via an API) at a stable URL, such as:
+
+> http://{your website}/opendata/projects/{project-id}.json
+
+If you are also publishing contracting data using OCDS, publish each new release of data as a JSON file at a stable URL such as:
+
+> http://{your website}/opendata/contracting/{ocid}/{release-id}.json
+
+Make sure your project level files include links in the `contractingProcess/releases` section to each related OCDS file.
+
+To make your data easier to access, consider providing:
+
+* A regularly updated bulk file of all your data for download
+* Flattened (spreadsheet or csv) representations of your data
+* A page on your website with details of how users can access your data
 
 ```eval_rst
-.. admonition:: Coming soon
-    :class: Note
+.. admonition:: Tip
+    :class: Tip
+
+    .. markdown::
+
+      [Flatten-tool](https://flatten-tool.readthedocs.io/en/latest/) can be used to convert between JSON and spreadsheet/csv format data.
```

Comment:
All-caps "CSV" throughout, unless you are specifically referring to the file extension, rather than the file format.

Reply:
Addressed in https://github.com/open-contracting/infrastructure/pull/46/commits/5c760acb02f31f25ee19ea29362dc314da9f633c